kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -278,21 +278,17 @@ object Fiber extends FiberPlatformSpecific:
             end State
             val state = new State
             import state.*
-            foreach(seq)((idx, io) => io.evalNow.foreach(v => state(Result.success(v))))
```

Comment:
I've removed this optimization that attempts to short-circuit the execution in case one of the computations is already done. The semantics of interruption wouldn't be consistent if we kept it. For example, in the test scenario "promise + plain value", it wouldn't interrupt the pending promise.